src/main/default/classes/TriggerHandler.cls

```diff
@@ -65,51 +73,66 @@ public class TriggerHandler {
 
         if (handlers != null && !handlers.isEmpty()) {
             for (HandlerInterface h : handlers) {
-                if (!handlerDisabled(h) && !objectHandlerDisabled(sObjectType, h)) {
+                if (!handlerDisabled(h)) {
                     h.handle();
                 }
             }
         }
     }
 
-    private static Set<String> disabledObjects = new Set<String>();
-    private static Set<String> disabledHandlers = new Set<String>();
-    private static Map<String, Set<String>> disabledObjectHandlers = new Map<String, Set<String>>();
-    @TestVisible
-    private static void cacheTriggerControl(Trigger_Control__mdt[] triggerControls) {
-        disabledObjects.clear();
-        disabledHandlers.clear();
-        disabledObjectHandlers.clear();
-        for (Trigger_Control__mdt triggerControl : triggerControls) {
-            String objectName = triggerControl.Object__c == null ? null : triggerControl.Object__c.toLowerCase();
-            String handlerName = triggerControl.Class__c == null ? null : triggerControl.Class__c.toLowerCase();
-            if (handlerName == null && objectName != null) {
-                disabledObjects.add(objectName);
-            } else if (handlerName != null && objectName == null) {
-                disabledHandlers.add(handlerName);
-            } else if (handlerName != null && objectName != null) {
-                if (!disabledObjectHandlers.containsKey(objectName)) {
-                    disabledObjectHandlers.put(objectName, new Set<String>());
-                }
-                disabledObjectHandlers.get(objectName).add(handlerName);
+    private void doAutomaticBinding() {
+        if (!triggerConfig.enabled) {
+            return;
+        }
+        for (String handlerName : triggerConfig.handlerOrder) {
+            TriggerConfigService.HandlerConfig handlerConfig = triggerConfig.handlers.get(handlerName);
+            // do not bind if auto bind is off or if handler is disabled
+            if (!handlerConfig.autoBind || !handlerConfig.enabled) {
+                continue;
+            }
+            Type t = getType(handlerConfig.className);
+            if (t == null) {
+                System.debug(handlerConfig.className + '? Looks like that handler class does not exist!');
+                continue;
+            }
+            HandlerInterface handlerClass = (HandlerInterface) t.newInstance();
```

Comment:
important, this will error if the type doesn't implement the handler interface, need to check for that

Reply:
Added a check